### ISSUE TITLE
Since version 0.85 recorder is not connecting to mqtt anymore

### ISSUE THREAD

**Issue by ramonvanbiljouw** [owntracks/recorder#339]
Hi,

Since version 0.85 (March 9, 2020) the ot-recorder is not connecting to the MQTT Broker anymore. I have both the recorder and the broker running in a docker container on a Synology. 

The error message on the recorder side is: MQTT connection: rc=19 [Unknown error.] (errno=2; No such file or directory). Sleeping...
The error message on the MQTT Broker side is: Socket error on client <unknown>, disconnecting.

For the recorder side I have the following setup (relevant settings only, these settings are set using the Synology Docker user interface):
OTR_USER='recorder'
OTR_PASS='mypassword'
OTR_HOST=10.0.1.3
OTR_PORT=1883

And I have the following settings in the file: mosquitto.conf

```
pid_file /var/run/mosquitto.pid

persistence true
persistence_location /mqtt/data/
persistence_file mosquitto.db

user mosquitto

password_file /mqtt/config/passwd_file
allow_anonymous true

# Plain MQTT protocol
listener 1883
protocol mqtt

listener 8883
cafile /mqtt/config/certs/ca.crt
certfile /mqtt/config/certs/Apple-Server.crt
keyfile /mqtt/config/certs/Apple-Server.key
tls_version tlsv1.2

# End of plain MQTT configuration

# Plain WebSockets configuration
listener 9001
protocol websockets
http_dir /mqtt/owntracks-cards

# End of plain Websockets configuration

log_dest file /mqtt/log/mosquitto.log
log_dest stdout
```

These setting worked perfectly until version 0.85. I hope someone is able to help me. 
Thank you in advance.

Regards, Ramon

**Comment by jpmens** [comment]
Sorry to hear you are experiencing problems. I've just verified and I don't see we've changed anything in the connection handling.

We're going to need your help in diagnosing this a bit further; can you give us the last, say 30 lines of log of each of the Recorder and Mosquitto? You are running on RFC1918 private IP addresses, so there's not much you have to hide. :) 

Another question: what happens when you downgrade back to what you had prior to 0.8.5?

**Comment by mbhoek** [comment]
I'm experiencing the same error in a similar environment (using Docker containers). 
```
owntracks_1       | * 10:18:25 owntracks/ping/ping                 t=  tid=pp loc=XXX,YYY []  (gcpuzg2)
owntracks_1       | ot-recorder[7]: http: POST /api/0/last
owntracks_1       | ot-recorder[7]: Disconnected. Reason: 0x13 [unknown reason]
owntracks_1       | ot-recorder[7]: MQTT connection: rc=19 [Unknown error.] (errno=2; No such file or directory). Sleeping...
mosquitto_1       | 1584181144: New connection from 192.168.16.3 on port 1883.
mosquitto_1       | 1584181144: Client <unknown> disconnected due to protocol error.
owntracks_1       | ot-recorder[7]: http: POST /pub
owntracks_1       | ot-recorder[7]: http: POST /api/0/last
```

The docker containers used are
- eclipse-mosquitto:latest
- owntracks/recorder:latest 

I've confirmed that the same error occurs when using eclipse-mosquitto 1.6.9, 1.6.8 and 1.6.7.
I've confirmed that the same error occurs when using owntracks/recorder 0.8.6-8 and 0.8.5-7.

**The error does not occur when using owntracks/recorder 0.8.3-6.**

Apparently 0.8.4 was never released as a Docker image, so I can only [compare 0.8.3 to 0.8.5](https://github.com/owntracks/recorder/compare/0.8.3...0.8.5), but I see no obvious causes there for the problem. This led me to believe that there might be an issue with one of the lib dependencies, so I tried adding `apk update` to the `docker-recorder` Dockerfile, but that didn't help either. Adding the `--debug` flag did not reveal any additional information afaict.

Please let me know if you need any additional information.

TL:DR: the latest working Docker image is `owntracks/recorder:0.8.3-6`.

**Comment by jpmens** [comment]
@mbhoek thank you for your very good report. There's one thing only I see which could have the effect we see in the log you posted: `(errno=2; No such file or directory)`

In #336 we added support for honouring `OTR_CAPATH`, and I wonder whether that value is set somewhere in either your environment or in the Defaults file. Could you check that for me?

(Unfortunately I'm not in a position to test this currently which is why I need your assistance.)

**Comment by mbhoek** [comment]
@jpmens It was not defined in my environment, but I did notice that the default `recorder.conf` contains `OCR_CAPATH = "/config"` ([line 8](https://github.com/owntracks/docker-recorder/blob/26cdb7714fc97d9fb73761c5c5d43f0a407f7f44/recorder.conf#L8)), a path that I don't use. If I comment out that line, it works fine (I don't use certificates so I'm not sure if it will affect other areas). Fixed in owntracks/docker-recorder#34.

**Comment by jpmens** [comment]
@mbhoek thank you very much. I have built and pushed a new image `owntracks/recorder:0.8.6-9`

**Comment by mbhoek** [comment]
@jpmens I can confirm both `owntracks/recorder:0.8.6-9` and `owntracks/recorder:latest` no longer have the issue. Thanks for the quick turnaround!

**Comment by jpmens** [comment]
Your investigaion made this possible; thanks for your feedback.  :-)
